pkg/s3wrapper/client.go

```diff
@@ -473,52 +476,94 @@ func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]st
 
 func (c *S3Client) UploadBootFiles(ctx context.Context, openshiftVersion string) error {
 	rhcosImage, err := c.versionsHandler.GetRHCOSImage(openshiftVersion)
-
 	if err != nil {
 		return err
 	}
 
-	return c.uploadBootFiles(ctx, c.GetBaseIsoObject(openshiftVersion), rhcosImage)
+	baseIsoObject := c.GetBaseIsoObject(openshiftVersion)
+	minimalIsoObject := isoeditor.GetMinimalIsoObject(openshiftVersion)
+	return c.uploadBootFiles(ctx, baseIsoObject, minimalIsoObject, rhcosImage, openshiftVersion)
 }
 
-func (c *S3Client) uploadBootFiles(ctx context.Context, isoObjectName, isoURL string) error {
+func (c *S3Client) uploadBootFiles(ctx context.Context, isoObjectName, minimalIsoObject, isoURL, openshiftVersion string) error {
 	log := logutil.FromContext(ctx, c.log)
 
-	exist, err := c.DoAllBootFilesExist(ctx, isoObjectName)
+	baseExist, err := c.DoAllBootFilesExist(ctx, isoObjectName)
+	if err != nil {
+		return err
+	}
+	minimalExist, err := c.DoesIsoFileExist(ctx, minimalIsoObject)
 	if err != nil {
 		return err
 	}
-	if exist {
+	if baseExist && minimalExist {
 		return nil
 	}
 
 	log.Infof("Starting Base ISO download for %s", isoObjectName)
-	tmpfile, err := DownloadURLToTemporaryFile(isoURL)
+	baseIsoPath, err := DownloadURLToTemporaryFile(isoURL)
 	if err != nil {
 		log.Error(err)
 		return err
 	}
-	defer os.Remove(tmpfile)
+	defer os.Remove(baseIsoPath)
 
 	exists, err := c.DoesPublicObjectExist(ctx, isoObjectName)
 	if err != nil {
 		return err
 	}
 	if !exists {
-		err = c.UploadFileToPublicBucket(ctx, tmpfile, isoObjectName)
+		err = c.UploadFileToPublicBucket(ctx, baseIsoPath, isoObjectName)
 		if err != nil {
 			return err
 		}
 		log.Infof("Successfully uploaded object %s", isoObjectName)
 	}
 
-	return ExtractBootFilesFromISOAndUpload(ctx, log, tmpfile, isoObjectName, isoURL, c)
+	if !baseExist {
+		err = ExtractBootFilesFromISOAndUpload(ctx, log, baseIsoPath, isoObjectName, isoURL, c)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !minimalExist {
+		rootfsURL := isoeditor.GetRootfsURL(c.cfg.ServiceBaseURL, openshiftVersion)
+		editor := isoeditor.CreateEditor(baseIsoPath, rootfsURL, log)
+		log.Infof("Extracting rhcos ISO (%s)", baseIsoPath)
+		minimalIsoDir, err := editor.CreateMinimalISOTemplate(ctx, openshiftVersion)
+		if err != nil {
+			log.Errorf("Error extracting rhcos ISO (%v)", err)
+			return err
+		}
+		defer os.Remove(minimalIsoDir)
+
+		// upload the minimal iso
+		minimalIsoPath := filepath.Join(minimalIsoDir, minimalIsoObject)
```

Comment:
Why not have the iso editor tell you the full path to the file? There's no reason it has to be the same as the object name.

Reply:
We already use GetMinimalISOObjectName internally in the editor for the iso name. We can have a getter in the editor instead to build the path.